### PR TITLE
Ingest Roads and OSM Rasters from Google Storage

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[OSM pipeline](https://trello.com/c/ka7OwdSB/29-osm-pipeline)

File: src/task.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        self._prep_asset_id(ee_dir, image_collection=False)
+
+        image_asset_ids = []
+        for image_uri in image_uris:
+            image_asset_id = f"{PROJECTS}/{self.ee_project}/{ee_dir}/{Path(os.path.splitext(image_uri)[0]).name}"
+            task_id = self._cp_storage_to_ee_image(image_uri, image_asset_id)
+            self.ee_tasks[task_id] = {}
+            image_asset_ids.append(image_asset_id)
+
+        self.wait()
```
Comment (saanobhaai) [inline src/task.py:132]: Is this necessary anymore?

File: src/task.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        bands = ee.List(list(metadata["bands"].values()))
+        projection = ee.Image(image_asset_ids[0]).projection()
+
+        def band_merge(attr_tag_meta):
+            atm = ee.Dictionary(attr_tag_meta)
+            _bands = ee.List(atm.get("bands"))
+            attribute = atm.get("attribute")
+            tag = atm.get("tag")
+
+            band_indices = _bands.map(lambda x: ee.Number(x).subtract(ee.Number(1)))
```
Comment (saanobhaai) [inline src/task.py:156]: I'm not sure lambdas are serializable and therefore that this behaves as you're intending

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Just a couple of questions...

File: src/task.py (hunk 1 above)
Comment (gridcell) [reply src/task.py:132]: that one is still necessary to wait until all the split images are loaded before grouping them

File: src/task.py (hunk 2 above)
Comment (saanobhaai) [reply src/task.py:156]: Never mind -- turns out lambdas are fine. 

In which case this PR looks good to merge to me...

**Pull request merged**